invenio_config/__init__.py

```diff
@@ -22,19 +22,148 @@
 # waive the privileges and immunities granted to it by virtue of its status
 # as an Intergovernmental Organization or submit itself to any jurisdiction.
 
-"""Invenio configuration module."""
+"""Invenio configuration module.
+
+Invenio-Config is a core component of Invenio responsible for loading
+configuration from various sources to application instance. The loader is used
+to merge these sources in predefined order ensuring correct behavior in
+all common scenarios.
```

Comment:
It would be good to provide some minimal context for people landing on this page as a casual visitor, for example:

```
Invenio-Config is a _base package_ of the Invenio digital library framework. 
It is usually installed automatically as a dependency.
```

linking to a place like http://invenio.readthedocs.io/en/latest/modules/index.html